SAP2000_Adapter/CRUD/Create/Material.cs

```diff
@@ -56,38 +58,125 @@ private bool CreateObject(IMaterialFragment material)
             else if (m_model.PropMaterial.SetMaterial(bhName, matType, color, notes, guid) == 0) //create the material
             {
                 sap2000id.Id = bhName;
-
-                if (material is IIsotropic)
-                {
-                    IIsotropic isotropic = material as IIsotropic;
-                    if (m_model.PropMaterial.SetMPIsotropic(bhName, isotropic.YoungsModulus, isotropic.PoissonsRatio, isotropic.ThermalExpansionCoeff) != 0)
-                        CreatePropertyWarning("Isotropy", "Material", bhName);
-
-                }
-                else if (material is IOrthotropic)
-                {
-                    IOrthotropic orthoTropic = material as IOrthotropic;
-                    double[] e = orthoTropic.YoungsModulus.ToDoubleArray();
-                    double[] v = orthoTropic.PoissonsRatio.ToDoubleArray();
-                    double[] a = orthoTropic.ThermalExpansionCoeff.ToDoubleArray();
-                    double[] g = orthoTropic.ShearModulus.ToDoubleArray();
-                    if (m_model.PropMaterial.SetMPOrthotropic(bhName, ref e, ref v, ref a, ref g) != 0)
-                        CreatePropertyWarning("Orthotropy", "Material", bhName);
-                }
-
-                if (m_model.PropMaterial.SetWeightAndMass(bhName, 2, material.Density) != 0)
-                    CreatePropertyWarning("Density", "Material", bhName);
+                success &= SetObject(material);
```

Comment:
We can ignore the output of SetObject- if the object is created (even with no properties), then CreateObject() should return true. SetObject should only raise issues for the user